src/components/PhotoFrame.tsx

```diff
@@ -454,9 +533,32 @@ const PhotoFrame = ({
                     } else {
                         url = await DownloadManager.getFile(item, true);
                     }
-                    galleryContext.files.set(item.id, url);
+                    if (item.metadata.fileType !== FILE_TYPE.LIVE_PHOTO) {
+                        galleryContext.files.set(item.id, url);
+                    }
+                }
+                if (item.metadata.fileType === FILE_TYPE.LIVE_PHOTO) {
+                    let imageURL;
+                    let videoURL;
+                    if (galleryContext.files.has(item.id)) {
+                        [imageURL, videoURL] = galleryContext.files
+                            .get(item.id)
+                            .split(',');
+                    } else {
+                        ({ imageURL, videoURL } = await splitLivePhoto(
```

Comment:
we can have this logic directly in the `convertForPreview` function? 

why send the zip blob and then do the call the conversion logic here?